backend/pkg/modules/internal/fuzzer/tools/specs.go

```diff
@@ -1,36 +1,163 @@
 	"fmt"
+	"net/http"
+
+	"github.com/getkin/kin-openapi/openapi2"
+	"github.com/getkin/kin-openapi/openapi2conv"
+	"github.com/getkin/kin-openapi/openapi3"
+	"github.com/getkin/kin-openapi/routers"
+	"k8s.io/utils/strings/slices"
 
 	"github.com/openclarity/apiclarity/api/server/models"
 	"github.com/openclarity/apiclarity/backend/pkg/modules/internal/core"
+	"github.com/openclarity/apiclarity/backend/pkg/modules/internal/fuzzer/logging"
+	"gopkg.in/yaml.v2"
 )
 
-func GetAPISpecsInfo(ctx context.Context, accessor core.BackendAccessor, apiID uint) (*models.OpenAPISpecs, error) {
+// FuzzerSpecInfo An object containing info about a spec.
+type FuzzerSpecsInfo struct {
+	ProvidedSpec          string
+	ReconstructedSpec     string
+	ProvidedSpecInfo      *models.SpecInfo
+	ReconstructedSpecInfo *models.SpecInfo
+}
+
+func GetAPISpecsInfo(ctx context.Context, accessor core.BackendAccessor, apiID uint) (*FuzzerSpecsInfo, error) {
+	fuzzerSpecsInfo := FuzzerSpecsInfo{}
+
 	apiInfo, err := accessor.GetAPIInfo(ctx, apiID)
 	if err != nil {
 		return nil, fmt.Errorf("can't access to apiID (%v), error=%v", apiID, err)
 	}
 
-	specsInfo := &models.OpenAPISpecs{}
 	if apiInfo.ProvidedSpecInfo != "" {
 		specInfo := models.SpecInfo{}
 		if err := json.Unmarshal([]byte(apiInfo.ProvidedSpecInfo), &specInfo); err != nil {
 			return nil, fmt.Errorf("failed to unmarshal provided spec info. info=%+v: %v", apiInfo.ProvidedSpecInfo, err)
 		}
-		specsInfo.ProvidedSpec = &specInfo
+		fuzzerSpecsInfo.ProvidedSpecInfo = &specInfo
+	}
+
+	if apiInfo.ProvidedSpec != "" {
+		fuzzerSpecsInfo.ProvidedSpec = apiInfo.ProvidedSpec
 	}
 
 	if apiInfo.ReconstructedSpecInfo != "" {
 		specInfo := models.SpecInfo{}
 		if err := json.Unmarshal([]byte(apiInfo.ReconstructedSpecInfo), &specInfo); err != nil {
 			return nil, fmt.Errorf("failed to unmarshal reconstructed spec info. info=%+v: %v", apiInfo.ReconstructedSpecInfo, err)
 		}
-		specsInfo.ReconstructedSpec = &specInfo
+		fuzzerSpecsInfo.ReconstructedSpecInfo = &specInfo
+	}
+
+	if apiInfo.ReconstructedSpec != "" {
+		fuzzerSpecsInfo.ReconstructedSpec = apiInfo.ReconstructedSpec
+	}
+
+	return &fuzzerSpecsInfo, nil
+}
+
+func IsV2Specification(data []byte) bool {
+	specV2OrV3 := struct {
+		Openapi *string `json:"openapi"`
+		Swagger *string `json:"swagger"`
+	}{}
+	if err := yaml.Unmarshal(data, &specV2OrV3); err != nil {
+		return false
+	}
+	return specV2OrV3.Swagger != nil
+}
+
+func IsV3Specification(data []byte) bool {
+	specV2OrV3 := struct {
+		Openapi *string `json:"openapi"`
+		Swagger *string `json:"swagger"`
+	}{}
+	if err := yaml.Unmarshal(data, &specV2OrV3); err != nil {
+		return false
+	}
+	return specV2OrV3.Openapi != nil
+}
+
+func LoadSpec(spec []byte) (*openapi3.T, error) {
+	if IsV2Specification(spec) {
+		logging.Logf("[Fuzzer] getDocFromSpec(): spec V2 identified")
+		loader := openapi3.NewLoader()
+		var docV2 openapi2.T
+		if err := json.Unmarshal(spec, &docV2); err != nil {
+			panic(err)
```

Comment:
better not to use `panic` and return the error back for better error handling.
(true for all files)

Reply:
You totally right. I use 'panic' only at first stage of implementation, don't know why I skip this file when I took them off. 
Done.